sentinelhub/data_utils.py

```diff
@@ -130,23 +138,42 @@ def _get_failed_intervals(
         if "error" in interval:
             failed_intervals.append((interval["interval"]["from"], interval["interval"]["to"]))
 
-    return {"identifier": identifier, "failed_intervals": failed_intervals} if failed_intervals else None
+    return failed_intervals
 
 
-def get_failed_statistical_requests(result_data: List[JsonDict]) -> List[Dict[str, Union[str, List[Tuple[str, str]]]]]:
-    """Collect failed requests of (Batch) Statistical Results
+def _get_failed_batch_statistical_requests(result_data: List[JsonDict]) -> List[JsonDict]:
+    """Collect failed Batch Statistical requests
 
-    :param result_data: An input representation of (Batch) Statistical API result.
-    :return: Failed requests of (Batch) Statistical Results.
+    :param request_data: An input representation of Batch Statistical API result
+        returned from `AwsBatchStatisticalResults.get_data()`.
+    :return: Failed Batch Statistical requests.
     """
     failed_requests = []
     for result in result_data:
-        identifier, response = result["identifier"], result["response"]
-        if not response:
-            failed_requests.append({"identifier": identifier})
-        else:
-            failed_intervals = _get_failed_intervals(identifier, response["data"])
-            if failed_intervals:
-                failed_requests.append(failed_intervals)
+        identifier = result["identifier"]
+        if "error" in result or not result["response"]:
+            failed_requests.append({"identifier": identifier, "failed_intervals": "full time range"})
+            continue
+
+        failed_intervals = _get_failed_intervals(result["response"]["data"])
+        if failed_intervals:
+            failed_requests.append({"identifier": identifier, "failed_intervals": failed_intervals})
 
     return failed_requests
+
```

Comment:
what if `_get_failed_intervals` instead accepted a full result` and returned `"full time range"` in case there is an error?
Then you could perhaps do something like: 
```python
failed_intervals = [{"identifier": identifier, "failed_intervals": _get_failed_intervals(result)} for result in resultdata]
return [failure_report for failure_report in failed_intervals if failure_report["failed_intervals"] is not None]
```